test/boost/database_test.cc

```diff
@@ -715,10 +715,11 @@ SEASTAR_TEST_CASE(snapshot_list_okay) {
         BOOST_REQUIRE_EQUAL(sd.live, 0);
         BOOST_REQUIRE_GT(sd.total, 0);
 
-        lister::scan_dir(table_dir(cf), lister::dir_entry_types::of<directory_entry_type::regular>(), [] (fs::path parent_dir, directory_entry de) {
-            fs::remove(parent_dir / de.name);
-            return make_ready_future<>();
-        }).get();
+        auto table_directory = table_dir(cf);
+        auto files = collect_files(table_directory).get();
+        for (auto& f : files) {
```

Comment:
nitpick: there is no real need for the temporary variable (that lives on beyond the loop). You can do it as:
```c++
for (auto& f : collect_files(table_directory).get()) {
  fs::remove(table_directory / f);
}
```